src/vstest.console/Processors/RunSettingsArgumentProcessor.cs

```diff
@@ -137,6 +138,12 @@ public void Initialize(string argument)
                 this.runSettingsManager.AddDefaultRunSettings();
 
                 this.commandLineOptions.SettingsFile = argument;
+
+                if (this.runSettingsManager.QueryRunSettingsNode("RunConfiguration.EnvironmentVariables") != null)
+                {
+                    this.commandLineOptions.InIsolation = true;
+                    this.runSettingsManager.UpdateRunSettingsNode(InIsolationArgumentExecutor.RunSettingsPath, "true");
```

Comment:
Just curious, why is this required if the CLI option is already set.

Reply:
[https://github.com/microsoft/vstest/blob/master/src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs#L244](https://github.com/microsoft/vstest/blob/master/src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs#L244) This checks inisolation node from runsettings.